home work 6

bmv87 opened the pull request.

bmv87 committed:
feat: hw-6

FlameXander commented on src/main/java/ru/otus/java/basic/homework/hw6/animals/Cat.java:
```diff
@@ -0,0 +1,33 @@
+        this.bellyful = false;
+    }
+
+    private boolean bellyful;
```
Поля лучше объявлять "пачкой", чтобы не искать их среди геттеров/сеттеров

bmv87 replied:
Исправлено

FlameXander commented on src/main/java/ru/otus/java/basic/homework/hw6/bowls/Bowl.java:
```diff
@@ -0,0 +1,36 @@
+        }
+    }
+
+    public boolean getFood(int foodVolume) {
```
Очень странное название метода. Не глядя в код покажется что это какой-то странный геттер с аргументом. Надо бы поменять на что-нибудь вроде `reduceFood(int amount);`

bmv87 replied:
Переименовано

FlameXander commented on src/main/java/ru/otus/java/basic/homework/hw6/bowls/Bowl.java:
```diff
@@ -0,0 +1,36 @@
+            System.out.printf("Неправильно задан объем еды: %d ед.", foodVolume);
+            return false;
+        }
+        if (currentFoodVolume < foodVolume) {
```
А если еды прям ровно хватило, почему false?

bmv87 replied:
Тут так и есть, если currentFoodVolume == foodVolume, то в условие не входит и далее возвращает true.
currentFoodVolume - запас еды
foodVolume - запрошенное количество еды

FlameXander replied:
Ой, да, эт я чего-то куда-то не туда посмотрел) Вы правы

FlameXander commented on src/main/java/ru/otus/java/basic/homework/hw6/bowls/Bowl.java:
```diff
@@ -0,0 +1,36 @@
+
+    public void putFood(int foodVolume) {
+        if (foodVolume < 0) {
+            System.out.printf("Неправильно задан объем еды: %d ед.", foodVolume);
```
Если такую проверку сделали, то почему не останавливаете работу метода? Ведь если подать -100, то в тарелке станет меньше еды?

FlameXander commented on src/main/java/ru/otus/java/basic/homework/hw6/animals/Cat.java:
```diff
@@ -0,0 +1,33 @@
+        this.appetite = appetite;
+    }
+
+    public boolean eatFromBowl(Bowl bowl) {
```
Мне кажется спокойно можно метод назвать `eat`

bmv87 replied:
Переименовано

bmv87 committed:
feat: refactoring after PR comments https://github.com/bmv87/otus-java/pull/6#discussion_r1605592949 https://github.com/bmv87/otus-java/pull/6#discussion_r1605593402 https://github.com/bmv87/otus-java/pull/6#discussion_r1605594457
fix: putFood validation in Bowl class after PR comment https://github.com/bmv87/otus-java/pull/6#discussion_r1605594259
